### PR TITLE
tests: reworked sometimes failing tests

### PR DESCRIPTION
**Pull request by mxschmitt**
Sometimes the following tests are failing:

https://github.com/react-bootstrap/react-bootstrap/runs/248586391#step:8:591

```
  <OverlayTrigger>
    ✖ Should pass transition callbacks to Transition
      Firefox 69.0.0 (Mac OS X 10.14.0)
    uncaught exception: AssertionError: expected 'next' to equal 'prev' (:0)

  ✖ "after each" hook for "allows only ProgressBar in children"
    Firefox 69.0.0 (Mac OS X 10.14.0)
  console.error.expected is undefined
  ./test/index.js/<@webpack:///test/index.js:63:1 <- test/index.js:227223:7
  

  ✖ "after each" hook for "should contain `embed-responsive` class"
    Firefox 69.0.0 (Mac OS X 10.14.0)
  console.error.expected is undefined
  ./test/index.js/<@webpack:///test/index.js:63:1 <- test/index.js:227223:7
```

https://github.com/react-bootstrap/react-bootstrap/runs/248886920#step:8:483

```
  <Modal>
    × Should pass transition callbacks to Transition
      Firefox 69.0.0 (Windows 10.0.0)
    uncaught exception: AssertionError: expected 2 to equal 0 (:0)
```

https://github.com/react-bootstrap/react-bootstrap/runs/248600092#step:8:620

```
  <SplitButton>
    ✖ should not invoke onClick when SplitButton.Toggle is clicked (prop)
      HeadlessChrome 77.0.3865 (Mac OS X 10.14.6)
    Error: Uncaught AssertionError: expected 'next' to equal 'prev' (node_modules/chai/chai.js:239)

  ✖ "after each" hook for "Should pass disabled to both buttons"
    HeadlessChrome 77.0.3865 (Mac OS X 10.14.6)
  TypeError: Cannot read property 'length' of undefined
      at Context.<anonymous> (webpack:///test/index.js:63:1 <- test/index.js:227223:54)
```

### CODE REVIEW

**Comment by mxschmitt** [comment]
The strange this is, that on macOS / Firefox not all tests are running (sometimes): https://github.com/react-bootstrap/react-bootstrap/pull/4593/checks?check_run_id=248579982

File: test/index.js [hunk 1]
```diff
@@ -5,8 +5,8 @@ import Adapter from 'enzyme-adapter-react-16';
 
 Enzyme.configure({ adapter: new Adapter() });
 
-function assertLength(length) {
-  return function $assertLength(selector) {
+const assertLength = length =>
```
Comment (taion) [inline test/index.js]: i think it's slightly nicer to have normal functions at top level... shades of https://github.com/airbnb/javascript/tree/master/react#class-vs-reactcreateclass-vs-stateless in preferring normal functions over arrows

**Comment by bpas247** [comment]
> The strange this is, that on macOS / Firefox not all tests are running (sometimes): https://github.com/react-bootstrap/react-bootstrap/pull/4593/checks?check_run_id=248579982

Should we open an issue to track this?

**Comment by mxschmitt** [comment]
> > The strange this is, that on macOS / Firefox not all tests are running (sometimes): https://github.com/react-bootstrap/react-bootstrap/pull/4593/checks?check_run_id=248579982
> 
> Should we open an issue to track this?

Yep let's do this. It's very randomly.

**Comment by taion** [comment]
I think GitHub cancels some of them automatically if one of them fails.

**Pull request merged**